### ISSUE TITLE
Validation producing boxes,scores, labels all -1

### ISSUE THREAD

**Issue by sshleifer** [fizyr/keras-retinanet#702]
I am encountering failure to evaluate on a custom dataset and have tried to debug quite a bit. Here are some clues. 

1) I'm on a cloud Ubuntu instance with reasonable versions: Keras==2.2.2, tensorflow==1.7.0.

2) `train.py` seems to work. The losses end up at 
`10000/10000 [==============================] - 2493s 249ms/step - los      s: 0.9290 - regression_loss: 0.5230 - classification_loss: 0.4060` after 1 epoch.
Then it runs validation on 2481 images and says mAP 0.000 and saves the model.

3) `debug.py` :cannot connect to X server. Probably because I can't make a display on my cloud machine. Is there a way to run this in a jupyter notebook?

4) When I convert the model in predict mode and load it into jupyter, and run:
`boxes, scores, labels = model.predict_on_batch(np.expand_dims(image, axis=0))`
(following the example notebook). all the boxes are `[-1, -1, -1, -1]` (-1.28 after scaling), all the scores are -1 and all the labels are -1 for all the images I have tried.

```
from keras_retinanet.utils.image import read_image_bgr
path =  '/home/paperspace/keras-retinanet/snapshots/resnet50_csv_01.h5' # same result after 3 epochs
#image = preprocess_image(image)
# image, scale = resize_image(image)
image = read_image_bgr(path)
boxes, scores, labels = model.predict_on_batch(np.expand_dims(image, axis=0))
print(labels.max(), boxes.max(), scores.max())
```

Note that adding or removing the commented lines doesn't effect the result.

### Example Image
![image](https://user-images.githubusercontent.com/6045025/46182981-bb6d7680-c29c-11e8-8bcb-d0e211dcd2f7.png)

Any idea what to try next/how to debug? Thanks!

**Comment by hgaiser** [comment]
I think @fernandocamargoti was working on the same dataset, perhaps he can help?

Did you start with COCO weights? My advice is to download the COCO weights and use those as starting point. Then let it train for a while (more than 1 epoch probably).

**Comment by hgaiser** [comment]
I'm closing this as it is not an issue with the code. Please use the Slack channel for further discussions.

**Comment by TimSmole** [comment]
@sshleifer I am facing same problem. Did you find a solution? Thanks!

**Comment by sshleifer** [comment]
I started with coco weights, as they suggested. Made things better.

**Comment by aman1902** [comment]
@sshleifer did you find a solution for this dataset. I've been trying to work but I'm stuck
